wilsonChan/assignments/mergeIntervals/lc57/lc57.py

```diff
@@ -1,18 +1,69 @@
 # Problem Statement #
 
-# Given a list of non-overlapping intervals sorted by their start time, insert a given interval at the correct position and merge all necessary intervals to produce a list that has only mutually exclusive intervals.
+# Given a list of non-overlapping intervals sorted by their start time, insert 
+# a given interval at the correct position and merge all necessary intervals 
+# to produce a list that has only mutually exclusive intervals.
+
+
+'''
+Given:
+-Array of intervals
+-Sorted by start time
+-New interval to insert into existing intervals
+
+Result:
+-Merge the intervals
+
+
+EXAMPLE:
+Original: [[1, 3], [5, 7], [8, 12]]        Insert: [4, 6]
+
+[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+ 1-----3
+             5-----7
+                      8-------------12
+
+          4-----6
+
+Output: [1,3] [4,7] [8,12]
+
+
+STEPS:
+1) Iterate through given intervals with for loop
+    -Compare new_interval variable with i (other elements/intervals witin array)
+    -Edge Cases:
+      -New interval within the range of other intervals
+      -New interval's range is before and after other intervals
```

Comment:
This approach is missing some key steps to help me understand what your code will do.